documentation/developers/docstring/README.md

```diff
@@ -0,0 +1,5913 @@
+             ===              ===  ===            ===
+
+Attention:
+* the circuit is constantly draining the battery! (leak current up to: 2.1µA)
+* the time between sample needs to be a minimum 1sec with this high impedance voltage divider
+      don't use the continuous conversion method!
+
+
+<a id="components.gpio.gpioz.plugin"></a>
+
+# components.gpio.gpioz.plugin
+
+The GPIOZ plugin interface build all input and output devices from the configuration file and connects
+
+the actions and callbacks. It also provides a very restricted, but common API for the output devices to the RPC.
+That API is mainly used for testing. All the relevant output state changes are usually made through callbacks directly
+using the output device's API.
+
+
+<a id="components.gpio.gpioz.plugin.output_devices"></a>
+
+#### output\_devices
+
+List of all created output devices
+
+
+<a id="components.gpio.gpioz.plugin.input_devices"></a>
+
+#### input\_devices
+
+List of all created input devices
+
+
+<a id="components.gpio.gpioz.plugin.factory"></a>
+
+#### factory
+
+The global pin factory used in this module
+
+Using different pin factories for different devices is not supported
+
+
+<a id="components.gpio.gpioz.plugin.IS_ENABLED"></a>
+
+#### IS\_ENABLED
+
+Indicates that the GPIOZ module is enabled and loaded w/o errors
+
+
+<a id="components.gpio.gpioz.plugin.IS_MOCKED"></a>
+
+#### IS\_MOCKED
+
+Indicates that the pin factory is a mock factory
+
+
+<a id="components.gpio.gpioz.plugin.CONFIG_FILE"></a>
+
+#### CONFIG\_FILE
+
+The path of the config file the GPIOZ configuration was loaded from
+
+
+<a id="components.gpio.gpioz.plugin.ServiceIsRunningCallbacks"></a>
+
+## ServiceIsRunningCallbacks Objects
+
+```python
+class ServiceIsRunningCallbacks(CallbackHandler)
+```
+
+Callbacks are executed when
+
+* Jukebox app started
+* Jukebox shuts down
+
+This is intended to e.g. signal an LED to change state.
+This is integrated into this module because:
+
+* we need the GPIO to control a LED (it must be available when the status callback comes)
+* the plugin callback functions provide all the functionality to control the status of the LED
+* which means no need to adapt other modules
+
+
+<a id="components.gpio.gpioz.plugin.ServiceIsRunningCallbacks.register"></a>
+
+#### register
+
+```python
+def register(func: Callable[[int], None])
+```
+
+Add a new callback function :attr:`func`.
+
+Callback signature is
+
+.. py:function:: func(status: int)
+    :noindex:
+
+**Arguments**:
+
+- `status`: 1 if app started, 0 if app shuts down
+
+<a id="components.gpio.gpioz.plugin.ServiceIsRunningCallbacks.run_callbacks"></a>
+
+#### run\_callbacks
+
+```python
+def run_callbacks(status: int)
+```
+
+
+
+<a id="components.gpio.gpioz.plugin.service_is_running_callbacks"></a>
+
+#### service\_is\_running\_callbacks
+
+Callback handler instance for service_is_running_callbacks events.
+
+See :class:`ServiceIsRunningCallbacks`
```

Comment:
Should `:class:` be rendered? Or be a link?

compare change in rfid/reader/__init__.py #55
https://github.com/s-martin/RPi-Jukebox-RFID/blob/63ecd190ea7303a7c643b79615f07b34a2f81a4e/src/jukebox/components/rfid/reader/__init__.py#L55

Reply:
Not all occurrences of :class: are currently converted to # syntax